### ISSUE TITLE
ng6 serve error

### ISSUE THREAD

**Issue by thomasbird1984** [UltimateSoftware/ng6-cli#42]
I'm getting an error, while trying to serve the application using `ng6 serve` command. I'm on version 0.6.6 for the ng6-cli tools.

<img width="1204" alt="screen shot 2017-04-14 at 1 02 40 pm" src="https://cloud.githubusercontent.com/assets/1638025/25050021/cea848ce-2112-11e7-8ef9-9bb1db37a007.png">

When I searched the error `Check failed: !value_obj->IsJSReceiver() || value_obj->IsTemplateInfo()` I found this:

https://github.com/nodejs/node/issues/10659#issuecomment-270927364

When ran the updates it found a lot of out of date dependencies, after running the commands I got rid of the above error but then I was getting an error with the `extract-text-webpack-plugin`. Since updating from the version 1 to 2 the syntax had changed. I rewrote each instance of the usage of the `ExtractTextPlugin` to follow the new version 2 syntax but couldn't get things working right. I just kept getting the same `wrong syntax` error.

<img width="1146" alt="screen shot 2017-04-14 at 3 22 06 pm" src="https://cloud.githubusercontent.com/assets/1638025/25053600/36acfb8c-2126-11e7-9733-a238c29d5874.png">

Any help would be greatly appreciated.

**Comment by BenDiuguid** [comment]
Hey @thomasbird1984 Are you using webpack v1 or v2?

Because I believe `extract-text-webpack-plugin` [anything above v1.0.1](https://github.com/webpack-contrib/extract-text-webpack-plugin#install) is meant to work with webpack v2?

If you need to upgrade to webpack v2, you can look at the reference implementation for [ionic](https://github.com/UltimateSoftware/ng6-cli/blob/master/templates/app/ionic/template/webpack.config.js) or [web](https://github.com/UltimateSoftware/ng6-cli/blob/master/templates/app/web/template/webpack.config.js) in the current templates, webpack also has great docs on migrating from v1 to v2 [here](https://webpack.js.org/guides/migrating/)

Let us know if you need any help!

**Comment by thomasbird1984** [comment]
@BenDiuguid I checked my webpack version its on 2.4.1. I've been using the same version of ng6-cli for a while and never had issues and then I started getting this error. After some research I found people saying to upgrade the webpack and extract text plugin and thats when I got the new error. I'd be glad going back to any version that will work, I'm just not sure how the issues started from the beginning.

**Comment by BenDiuguid** [comment]
Hmm, yeah I've never seen this issue before. What version of node are you using?

**Comment by thomasbird1984** [comment]
node -> v7.9.0
npm  -> 4.2.0

**Comment by BenDiuguid** [comment]
Have you tried different versions? Idk if I have tried with latest node in a while, so that could be the issue. Try latest v6 or 7.6.0?

**Comment by thomasbird1984** [comment]
I'll give that a whirl!

**Comment by johnrjj** [comment]
Odd - Node 7.9 with the latest ng6 works on my machine

**Comment by thomasbird1984** [comment]
It's got to be something with dependency versions giving me the issues but I haven't updated anything in quite some time when I started getting the error. What version of node, npm, webpack etc are you using that works with node 7.9 and ng6?

**Comment by BenDiuguid** [comment]
Have you tried `rm -rf node_modules && npm install` yet?

**Comment by thomasbird1984** [comment]
I did and it didn't help I never commited my last changes to package.json and webpack.config.js, so that if I couldn't get a solution then I could just check out the original versions. I've rm -rf node_modules multiple times then reinstalled but I keep getting the first `Check failed: !value_obj->IsJSReceiver()`

**Comment by BenDiuguid** [comment]
Dang, it was worth a shot 😂

**Comment by thomasbird1984** [comment]
@BenDiuguid  This is my code base, I remove the node_modules directory, I feel like I've tried everything possible and still can't get it working: https://drive.google.com/file/d/0B-8P3RmI2z7UOUpyRVJvWHVWVms/view?usp=sharing

**Comment by johnrjj** [comment]
HI Tom,

I tried running your source, and was able to recreate your stack trace error. I then created a new project from the ng6 templates and it ran successfully. That would indicate there is something wrong with your project (and perhaps its dependencies). Can you try making a new project with ng6 and try running it?

**Comment by BenDiuguid** [comment]
So the issue has something to do with the version of `node-sass` that you are using. I upgraded `node-sass` to latest and the error disappeared.

Also incase you we're curious my thought process was this: this strange error seems like a native module (it's possible to write native modules which hook into V8's C++ layer) and I knew `node-sass` was a native module, so I upgraded it and the error went away.

Also when I initially ran `npm install` it gave me some 404 error when trying to pull down the prebuilt version of `node-sass`, but again upgrading solved this issue!

Let me know if it worked for you and we can close out this issue 👍
